content/security/docs/multiaccount.md

```diff
@@ -0,0 +1,86 @@
+
+| ![](./images/multi-account-eks.jpg) |
+|:--:|
+| In the above diagram, AWS RAM is used to share subnets from a cluster account into a workload account. Then workloads running in EKS pods use IRSA to assume a role in their workload account and consume their AWS resources. |
+
+ 
+### Implementing a Multi Workload Account Strategy for Multi Tenant Cluster
+
+#### Sharing Subnets With AWS Resource Access Manager 
+
+[AWS Resource Access Manager](https://aws.amazon.com/ram/) (RAM) allows you to share resources across AWS accounts. 
+
+If [RAM is enabled for your AWS Organization](https://docs.aws.amazon.com/ram/latest/userguide/getting-started-sharing.html#getting-started-sharing-orgs), you can share the VPC Subnets from the Cluster account to your workload accounts. This will allow AWS resources owned by your workload accounts, such as ElastiCache Clusters or RDS DBs to be deployed into the same VPC as your EKS cluster, and be consumable by the workloads running on your EKS cluster.
+
+To share a resource via RAM, open up RAM in the AWS console of the cluster account and select "Resource Shares" and "Create Resource Share". Name your Resource Share and Select the subnets you want to share. Select Next again and enter the 12 digit account IDs for the workload accounts you wish to share the subnets with, select next again, and click Create resource share to finish. After this step, the workload account can deploy resources into those subnets.
+
+RAM shares can also be created programmatically, or with infrastructure as code.
+
+### Accessing AWS API Resources with IAM Roles For Service Accounts
+ 
+[IAM Roles for Service Accounts (IRSA)](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) allows you to deliver temporary AWS credentials to your workloads running on EKS. IRSA can be used to get temporary credentials for IAM roles in the workload accounts from the cluster account. This allows your workloads running on your EKS clusters in the cluster account to consume AWS API resources, such as S3 buckets hosted in the workload account seemlessly, and use IAM authentication for resources like AWS RDS Databases or AWS EFS FileSystems. 
+
+AWS API resources and other Resources that use IAM authentication in a workload account can only be accessed by credentials for IAM roles in that same workload account, except where cross account access is capable and has been explicity enabled.
+
+#### Enabling IRSA for cross account access
+
+To enable IRSA for workloads in your Cluster Account to access resources in your Workload accounts, you first must create an IAM OIDC identity provider in your workload account. This can be done with the same procedure for setting up IRSA, except the Identity Provider will be created in the workload account: https://docs.aws.amazon.com/eks/latest/userguide/enable-iam-roles-for-service-accounts.html
```

Comment:
This is not the only way to assume a role in another account. A service account can be mapped to an IAM role that allows it to assume a role in another account.

Reply:
Yeah bad that is not scaleable, it puts pressure on the amount of available IAM roles in the cluster account and customers will quickly run up on this issue.

Reply:
After working with many customers hitting this problem of scale, IRSA is a much more practical solution.

Reply:
It's still using IRSA but the role the pod assumes allows it to assume a role in another account.

Reply:
Role chaining means that the app authors need to bring this into their code or config to account for, and it means in a decently busy AWS account they'll quickly run against the default IAM role per account quota.

It also means that customer code has access to a role session in a customer account, which of course has no default entitlements but it means their code could potentially access resources in that account, which is against the isolation objective we're trying to achieve.